src/objects/npcs/enemies/bear.py

```diff
@@ -0,0 +1,92 @@
+from enum import Enum
+
+from src.common.state_machine import StateMachine
+from src.common.enums import Direction
+from src.objects.npcs.enemy import Enemy
+from src.objects.agent import Agent
+from src.common.point import Point
+from src.actions.prepare_to_slam import PrepareToSlam
+from src.actions.attacks.slam import Slam
+
+HP = 100
+
+
+class Bear(Enemy):
+    """
+    Bear boss.
+    """
+
+    def __init__(self, scene: 'Scene', position: Point) -> None:
+        super().__init__(scene, position, HP, BearStateMachine(self))
+
+    def get_fields(self, position: Point) -> list['Object']:
+        return [position,
+                position + Point(0, 1),
+                position + Point(1, 1),
+                position + Point(1, 0)]
+
+    def tick(self) -> None:
+        self.machine.tick()
+
+    def slam(self):
+        Slam(self, [self.properties.position + Direction.NORTH.value,
+                    self.properties.position + Direction.WEST.value,
+                    self.properties.position + Direction.SOUTH.value,
+                    self.properties.position + Direction.EAST.value]).execute()
+
+    def prepare_to_slam(self):
+        PrepareToSlam(self).execute()
+
+    def is_there_agent_to_slam(self):
+        points_around = [self.properties.position + Direction.NORTH.value,
+                         self.properties.position + Direction.WEST.value,
+                         self.properties.position + Direction.SOUTH.value,
+                         self.properties.position + Direction.EAST.value]
+        for target_position in points_around:
+            if not self.scene._objects_position_map.get(target_position):
+                continue
+            for target in self.scene.get_objects_by_position(target_position):
+                if isinstance(target, Agent):
+                    return True
+        return False
+
+
+class State(Enum):
+    IDLE = 0
+    MOVE = 1
+    PREPARE_TO_SLAM = 2
+    SLAM = 3
+
+
+class BearStateMachine(StateMachine):
```

Comment:
You can define this class inside `Bear`, it's not going to be used with anything else.